### PR TITLE
Added catch for predictors that do not support summary() 

### LINKED ISSUE #110: Unclear Error Message
**Describe the bug**
I tried calling the summary of a trained pipeline but got a strange error message. That message didn't tell which module is not supported.

**To Reproduce**
Execute
```python

# coding: utf-8

###############################################################################

# FactorizationMachineBinaryClassifier

import numpy
from nimbusml import Pipeline, FileDataStream
from nimbusml.datasets import get_dataset
from nimbusml.decomposition import FactorizationMachineBinaryClassifier
from nimbusml.feature_extraction.categorical import OneHotVectorizer

# data input (as a FileDataStream)
path = get_dataset('infert').as_filepath()
data = FileDataStream.read_csv(path, sep=',',
                               numeric_dtype=numpy.float32,
                               names={0: 'row_num', 5: 'case'})
print(data.head())
#    age  case education  induced  parity  pooled.stratum  row_num  ...
# 0  26.0   1.0    0-5yrs      1.0     6.0             3.0      1.0  ...
# 1  42.0   1.0    0-5yrs      1.0     1.0             1.0      2.0  ...
# 2  39.0   1.0    0-5yrs      2.0     6.0             4.0      3.0  ...
# 3  34.0   1.0    0-5yrs      2.0     4.0             2.0      4.0  ...
# 4  35.0   1.0   6-11yrs      1.0     3.0            32.0      5.0  ...

# define the training pipeline
pipeline = Pipeline([
    OneHotVectorizer(columns={'edu': 'education'}),
    FactorizationMachineBinaryClassifier(feature=['induced', 'edu', 'parity'],
                                         label='case')
])

# train, predict, and evaluate
# TODO: Replace with CV
metrics, predictions = pipeline.fit(data).test(data, output_scores=True)

# print predictions
print(predictions.head())

#   PredictedLabel  Probability     Score
# 0             0.0     0.370519 -0.529990
# 1             0.0     0.420984 -0.318737
# 2             0.0     0.364432 -0.556180
# 3             0.0     0.380421 -0.487761
# 4             0.0     0.365351 -0.552214

# print evaluation metrics
print(metrics)

#        AUC  Accuracy  Positive precision  Positive recall  ...
# 0  0.609639  0.665323                   0                0  ...


# In[13]:


import nimbusml; print(nimbusml.__version__)
print(pipeline.steps[1])
pipeline.summary()


```

**Expected behavior**
It should say `FieldAwareFactorizationMachine` doesn't have summary function.

### PR DESCRIPTION
**Pull request by mstfbl**
Fixes #110 . Added a catch for the .summary() function in Pipeline.py for the FactorizationMachineBinaryClassifier.

### CODE REVIEW

File: src/python/nimbusml/tests/model_summary/test_model_summary.py [hunk 1]
```diff
@@ -96,7 +96,23 @@ def test_model_summary(self):
                 [OneHotVectorizer() << categorical_columns, learner])
             train_stream = FileDataStream(train_file, schema=file_schema)
             pipeline.fit(train_stream, label_column)
-            pipeline.summary()
+            # If Pipeline does contain FactorizationMachineBinaryClassifier,
+            # check for TypeError thrown by pipeline as 
+            # FactorizationMachineBinaryClassifier does not support summary()
```
Comment (ganik) [inline src/python/nimbusml/tests/model_summary/test_model_summary.py]: This is not specific to Factorization Machine. Any other trainer, transformer needs to be handled.

**Comment by mstfbl** [comment]
On the ML.NET end, if a summary model is not obtained from any trainers, an IDataView is returned in the following way (in line 89, Microsoft.ML.Data\EntryPoints\SummarizePredictor.cs:

>             if (dvGetter == null && rowGetter == null)
>             {
>                 var bldr = new ArrayDataViewBuilder(env);
>                 var summaryModel = predictor as ICanSaveSummary;
> 
>                 // Save a data view containing one row and one column with the model summary.
>                 if (summaryModel != null)
>                 {
>                     var sb = new StringBuilder();
>                     using (StringWriter sw = new StringWriter(sb))
>                         summaryModel.SaveSummary(sw, schema);
>                     bldr.AddColumn("Summary", sb.ToString());
>                 }
>                 else
>                     **bldr.AddColumn("PredictorName", predictor.GetType().ToString());**
>                 summary = bldr.GetDataView();
>             }

In the Python end, we receive a Pandas DataFrame with 1 column named `PredictorName` with the name of the given Predictor. Thus, unless the way of checking for the summary of a given model changes in ML.NET, we can check for the usability of the .summary() function in NimbusML by checking to see if the produced output equals this DataFrame.

**Comment by ganik** [comment]
PLs add test to show what FactorizationMachine error message is now

File: src/python/nimbusml/tests/model_summary/test_model_summary.py [hunk 2]
```diff
@@ -96,9 +96,11 @@ def test_model_summary(self):
                 [OneHotVectorizer() << categorical_columns, learner])
             train_stream = FileDataStream(train_file, schema=file_schema)
             pipeline.fit(train_stream, label_column)
-            pipeline.summary()
+            try:
+                pipeline.summary()
+            except:
+                assert False, ("Troubled learner: ", learner)
```
Comment (pieths) [inline src/python/nimbusml/tests/model_summary/test_model_summary.py]: Why was this modified? It appears that the extra code does not add any functionality.

File: src/python/nimbusml/tests/model_summary/test_model_summary.py [hunk 3]
```diff
@@ -107,6 +109,18 @@ def test_model_summary_not_supported(self):
...
+        data = FileDataStream.read_csv(path, sep=',',
+                               names={0: 'row_num', 5: 'case'})
+        pipeline = Pipeline([
+            OneHotVectorizer(columns={'edu': 'education'}),
+            FactorizationMachineBinaryClassifier(feature=['induced', 'edu', 'parity'],
+                                         label='case')
+        ])
+        pipeline.fit(data)
+        assert_raises(TypeError, pipeline.summary)
+
```
Comment (pieths) [inline src/python/nimbusml/tests/model_summary/test_model_summary.py]: Is this method necessary? It looks like it does the same test as `test_model_summary_not_supported`.

Comment (mstfbl) [reply src/python/nimbusml/tests/model_summary/test_model_summary.py]: I added this specific case of a test on Gani's request earlier to show what the FactorizationMachine error message is now.

**Comment by mstfbl** [comment]
@ganik I am now specifically checking for the error raised by the FactorizationMachine classifier in the following way:
https://github.com/microsoft/NimbusML/blob/208fe88d31d603c17953dbdacfbbeb281cf91704/src/python/nimbusml/tests/model_summary/test_model_summary.py#L119-L124

File: src/python/nimbusml/pipeline.py [hunk 4]
```diff
@@ -2581,6 +2581,10 @@ def summary(self, verbose=0, **params):
             self._run_time = time.time() - start_time
             raise e
 
+        # .summary() not supported if size of summary_data
+        # is less or equal to 1 (if only PredictedName in summary_data)
+        if summary_data.size == 1 and summary_data.columns.values == ["PredictorName"]:
+            raise TypeError("One or more transformers in this pipeline do not support the .summary() function.")
```
Comment (pieths) [inline src/python/nimbusml/pipeline.py]: Should "transformers" be "predictors" here?

Comment (mstfbl) [reply src/python/nimbusml/pipeline.py]: I was thinking they could be interchangeable in this case, but I believe you're right, "predictors" is more accurate here.
